digest/src/core_api/ct_variable.rs

```diff
@@ -5,17 +5,24 @@ use super::{
 use crate::HashMarker;
 #[cfg(feature = "mac")]
 use crate::MacMarker;
+#[cfg(feature = "oid")]
+use const_oid::{AssociatedOid, ObjectIdentifier};
 use core::{fmt, marker::PhantomData};
 use crypto_common::{
     generic_array::{ArrayLength, GenericArray},
     typenum::{IsLess, IsLessOrEqual, Le, LeEq, NonZero, U256},
     Block, BlockSizeUser, OutputSizeUser,
 };
 
+/// Dummy type used with [`CtVariableCoreWrapper`] in cases when
+/// resulting hash does not have a known OID.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub struct NoOid;
+
 /// Wrapper around [`VariableOutputCore`] which selects output size
 /// at compile time.
 #[derive(Clone)]
-pub struct CtVariableCoreWrapper<T, OutSize>
+pub struct CtVariableCoreWrapper<T, OutSize, O = NoOid>
```

Comment:
Instead of using an extra generic type here, could `CtVariableCoreWrapper` have an impl of `AssociatedOid` which delegates to `inner`, and is bounded on `inner` having an impl of `AssociatedOid`?

Reply:
If I understand you correctly, then no, it will not work. See the SHA-224 vs SHA-256 issue raised in my earlier comment. They have the same core (i.e. the same `inner` type), so with such solution they can not have different OIDs.

Reply:
Ugh, ok. That's really rather unfortunate.